third_party/ring/src/rand.rs

```diff
@@ -452,7 +452,11 @@ mod uefi {
                 static mut OPENSSL_ia32cap_P: [u32; 4];
             }
             const FLAG: u32 = 1 << 30;
-            unsafe { OPENSSL_ia32cap_P[1] & FLAG == FLAG }
+            // Somehow this check continues to require patching, regardless
+            // of qemu CPU config. :/. That is even though RDRAND is in fact
+            // available and working.
+            // unsafe { OPENSSL_ia32cap_P[1] & FLAG == FLAG }
+            true
```

Comment:
Status: 

QEMU with KVM can be started from the within the docker container (requires KVM to available on the host). RDRAND is then available in the UEFI app and gens random numbers. 

Oddly the check in ring for whether `RDRAND` continues to fail in all scenarios, even when RDRAND is in fact available. TODO: investigate why that's happening, clean up the implementation of running qemu/kvm in docker